synapse/rest/client/sync.py

```diff
@@ -239,29 +239,33 @@ async def encode_response(
         else:
             raise Exception("Unknown event format %s" % (filter.event_format,))
 
-        joined = await self.encode_joined(
-            sync_result.joined,
-            time_now,
-            access_token_id,
-            filter.event_fields,
-            event_formatter,
-        )
+        with start_active_span("sync.encode_joined"):
```

Comment:
Could we use `trace` on each of these functions or do you prefer the context manager approach?

Reply:
oh, good idea.

Reply:
done in fd7b6a0